### ISSUE TITLE
App freezes when pasting dup file

### ISSUE THREAD

**Issue by nandox** [vifm/vifm#468]
Vifm freezes when I try to copy or cut and then paste a file with the same name as a file already existing in the folder. The screen to choose whether to Replace/Rename/Cancel shows up but the app doesn't respond to any input.

Here is the info about the environment:
operating system: Linux raspberrypi 4.19.42-v7+ #1219 SMP Tue May 14 21:20:58 BST 2019 armv7l
version of curses library:  libncursesw5-dev/stable,now 6.0+20161126-1+deb9u2 armhf
terminal emulator: putty 0.72 for windows
color scheme: astrell

Vifm version info:
Version: 0.10.1
Git info: built out of repository
Compiled at: Aug 23 2019 13:48:37

Support of extended keys is on
Parsing of .desktop files is enabled
Without GTK+ library
Without magic library
Without X11 library
Without dynamic loading of X11 library
With file program
With -n option for cp and mv
With remote command execution

**Comment by xaizek** [comment]
I don't see it freezing and I'm not sure terminal emulator affects this. Do I get it right that the sequence to reproduce it looks something like below?

1. `:mkdir a b`
2. `:touch a/file b/file`
3. `:cd a b`
4. <kbd>yy</kbd>
5. <kbd>Space</kbd>
6. <kbd>p</kbd>
7. Further input is ignored.

**Comment by nandox** [comment]
Hi! Yes, those stops reproduce the issue. I also tried with different terminal emulators, Git Bash in Windows and the terminal that comes with Raspbian graphical interface, and it failed in the same way in both cases. Well, actually, after I followed the steps you suggested, I just need to get into Vifm for the terminal to stop working.

**Comment by nandox** [comment]
Correction: it worked in Raspbian terminal a couple of times, but now it doesn't work. I restarted the terminal and same issue now. Weird.

**Comment by xaizek** [comment]
> Well, actually, after I followed the steps you suggested, I just need to start Vifm for the terminal to stop working.

So, it hangs just when you open vifm with those two directories (`a` and `b`) loaded into panes? That would be even more puzzling.

**Comment by nandox** [comment]
Yes, it hanged there just after it failed with the steps you sent. As said before, to make it work again I just had to start vifm with a different starting folder. It looks to me that something was not compiled correctly in the RaspberryPi, even though I didn't get any error messages or warnings. But well, I'll update the Raspbian version in the Pi and then I'll compile it again, to see if that helps.

**Comment by nandox** [comment]
BTW, when I run the tests in the Pi all of them report success. Only 8 tests in the utils folder are "skipped", I don't know why. How do I figure out which ones are skipped and the reason?

**Comment by xaizek** [comment]
There are 8 Windows-specific tests in utils suite. That's fine.

Thanks for the bug report, I broke the dialog by introducing this check https://github.com/vifm/vifm/blob/224df4b048acd7c1ca10a7022b198f7a0a66be3c/src/modes/dialogs/msg_dialog.c#L129-L132
I did this while addressing warnings when building on ARM. The check should be something like (also not ideal, might be better to do the fix differently):
```c
if((int)key < (int)CHAR_MIN || (int)key > (int)CHAR_MAX) 
```
